src/Autofac/Builder/RegistrationBuilderResources.resx

```diff
@@ -120,4 +120,7 @@
   <data name="ComponentDoesNotSupportService" xml:space="preserve">
     <value>The type '{0}' is not assignable to service '{1}'.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="OnlyRegisterableTypesAllowed" xml:space="preserve">
+    <value>The type '{0}' is not concrete. Only concrete types (no interfaces, abstract classes, open generics, etc.) can be registered by type.</value>
```

Comment:
This is the exception message that will be thrown if you register types that generally can't be activated by reflection.